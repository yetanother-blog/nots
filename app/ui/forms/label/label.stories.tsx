import type { Meta, StoryObj } from '@storybook/react';
import { Label } from './label';

/**
 * The Label component is used to label an `input`, `select`, or `textarea`.
 * Nothing special here, but we want to stay DRY.
 *
 * Under the hood we inherit all types from
 * `LabelHTMLAttributes<HTMLLabelElement>` so we can handle this element as a
 * normal HTML `label` element. For further information, please refer to the
 * [MDN
 * documentation](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label).
 */
const meta: Meta<typeof Label> = {
  title: 'Forms/Label',
  component: Label,
};

export default meta;
type Story = StoryObj<typeof Label>;

export const Primary: Story = {
  args: {
    htmlFor: 'name',
    children: 'Name',
  },
};
