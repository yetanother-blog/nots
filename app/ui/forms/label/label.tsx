import type { FC, LabelHTMLAttributes, PropsWithChildren } from 'react';
import { Typography } from '~/ui/';

export type LabelProps = {
  className?: string;
} & PropsWithChildren &
  LabelHTMLAttributes<HTMLLabelElement>;

export const Label: FC<LabelProps> = ({
  className,
  children,
  htmlFor,
  ...labelProps
}) => {
  return (
    <Typography
      as="label"
      variant="body"
      textColor="grey-800"
      htmlFor={htmlFor}
      {...labelProps}
      className={className}
    >
      {children}
    </Typography>
  );
};
