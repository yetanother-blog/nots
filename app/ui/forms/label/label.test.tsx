import { render } from '@testing-library/react';
import { Label } from './label';

describe('Label', () => {
  it('should render label', () => {
    const { container } = render(<Label htmlFor="test">Label</Label>);
    expect(container).toMatchSnapshot();
  });

  it('should render with custom class', () => {
    const { container } = render(
      <Label htmlFor="test" className="custom-class">
        Label
      </Label>
    );
    expect(container).toMatchSnapshot();
  });
});
