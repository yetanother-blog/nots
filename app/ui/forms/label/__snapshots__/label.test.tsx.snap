// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Label > should render label 1`] = `
<div>
  <label
    class="antialiased font-sans text-base font-normal text-nots-grey-800 dark:text-white"
    for="test"
  >
    Label
  </label>
</div>
`;

exports[`Label > should render with custom class 1`] = `
<div>
  <label
    class="antialiased font-sans text-base font-normal text-nots-grey-800 dark:text-white custom-class"
    for="test"
  >
    Label
  </label>
</div>
`;
